### PR TITLE
feat(modules): ports outdated command

### PR DESCRIPTION
**Pull request by destifo**
add `ports outdated` command which shows the outdated installs in the env, with options to update specific installs or all at once.

### Describe your change

- [x] change `reducedRecipe` schema to hold `installSetId`s.
- [x] read from `recipe.json` and get current ports versions.
- [x] get latest versions from `PortBase` and log outdated ports table.
- [x] add `--update-no-confirm` and `--update-all` flags.
- [x] refactor and clean-up
- [x] write tests for subcommand and flags

### Motivation and context

[MET-485](https://linear.app/metatypedev/issue/MET-485/ports-outdated-command)

### Migration notes

_No Migrations Needed_

### Checklist

- [ ] The change come with new or modified tests
- [ ] Hard-to-understand functions have explanatory comments
- [ ] End-user documentation is updated to reflect the change

## Summary by CodeRabbit

- **New Features**
  - Added support for installing a plugin for CMake.
  - Introduced new functions for updating and installing dependencies.

- **Bug Fixes**
  - Improved error messages to include additional context information.
  - Updated dependency versions and platform specifications for better compatibility.

- **Improvements**
  - Enhanced environment context management.
  - Refined handling of provision types in POSIX environments.
  - Optimized the `PortsModule` class with new utility functions.

- **Documentation**
  - Updated type declarations and import statements for better clarity and maintainability.

### CODE REVIEW

File: modules/envs/types.ts [hunk 1]
```diff
@@ -92,4 +110,4 @@ export type WellKnownEnvRecipeX = zod.infer<
  */
 export type ProvisionReducer<P extends Provision> = (
   provisions: P[],
-) => Promise<WellKnownProvision[]>;
+) => Promise<WellKnownProvisionFat[]>;
```
Comment (Yohe-Am) [inline modules/envs/types.ts:113]: You don't need to go to this trouble of adding the Fat type. I'd keep the old provision type as is only focus on the semantics of the reducer. 
- Have the reducers return two arrays
- The first array is only allowed to have well known provisions
- The second will return any kind of reducer

The cook function will then only concern itself with the well known provisions but it'll still add the other provision types to the `recipe.json`.

Comment (Yohe-Am) [reply modules/envs/types.ts:113]: In fact, I'd not even bother with a second array. Return them all in one and the cook function can just ignore the ones with `ty` that it doesn't care about.

**Review by Yohe-Am — CHANGES_REQUESTED** [review]

File: modules/envs/types.ts (hunk 1 above)
Comment (Yohe-Am) [reply modules/envs/types.ts:113]: When trying to recover the InstallSet, you can go through the provisions and just pick out the `ghjk.ports.installSet` provision.

File: modules/envs/mod.ts [hunk 2]
```diff
@@ -50,10 +54,14 @@ export class EnvsModule extends ModuleBase<EnvsCtx, EnvsLockEnt> {
...
     const activeEnv = setEnv && setEnv != "" ? setEnv : config.defaultEnv;
 
-    return Promise.resolve({
+    const envsCtx = {
       activeEnv,
       config,
-    });
+    };
+
+    _ctx.blackboard.set(envsCtxBlackboardKey, envsCtx);
```
Comment (Yohe-Am) [inline modules/envs/mod.ts:62]: I'd follow the other blackboard state patterns there. Have get function create it if it doesn't exist. Blackboard call sites in processManifest know not to read shared values so it should be good.

Comment (destifo) [reply modules/envs/mod.ts:62]: but you might not have the data available to create the context the moment you are trying to access it.

Comment (Yohe-Am) [reply modules/envs/mod.ts:62]: You can create an empty config object without any envs in it the first time it's accessed.

Comment (destifo) [reply modules/envs/mod.ts]: Done.

File: modules/envs/types.ts (hunk 1 above)
Comment (destifo) [reply modules/envs/types.ts]: Also done.

**Review by Yohe-Am — APPROVED** [review]

**Comment by Yohe-Am** [comment]
@coderabbitai review

**Comment by zifeo** [comment]
@coderabbitai review

**Review by Yohe-Am — APPROVED** [review]
Good stuff. Go ahead a merge it on top of the envs PR.

**Pull request merged**